### ISSUE TITLE
Just updated (1.12.3) and yt-dlp no longer runs

### ISSUE THREAD

**Issue by gaussey** [holzschu/a-shell#734]
Output text is really difficult to select so screenshot.

![IMG_0998](https://github.com/holzschu/a-shell/assets/1004313/cac1b7f7-741a-470a-8559-9fd66b29b480)

I’ve tried installing with no dependencies and uninstalling and reinstalling a-shell to no avail.

This is on an M1 iPad, I’ll hold off upgrading my iPhone until I can find a solution. Luckily mini-a-shell still works.

**Comment by holzschu** [comment]
Thanks for the warning. I've run a quick check: the library *is* included, so the problem must be in `load_pycryptodome_raw_lib`). I'll have a more in-depth look.

**Comment by CrackerJackCheese** [comment]
yt-dlp still does not run as of 1.12.4, despite the attempted bug fixing.

**Comment by holzschu** [comment]
It works on both my test machines. Can you post a screenshot of the error message?

**Comment by CrackerJackCheese** [comment]
[
![IMG_8702](https://github.com/holzschu/a-shell/assets/67989956/09f89892-d487-464f-bbd5-499e1ca885e0)
![IMG_8703](https://github.com/holzschu/a-shell/assets/67989956/7e9bc781-c828-47fd-b176-adaf093b2d7f)

**Comment by holzschu** [comment]
Looking at the error message, I think you have installed `Cryptodome` using `pip`, but that installed the MacOS Arm64 binaries, and this version takes priority over the App version. You'll need to remove this version with `pip uninstall pycryptodome pycryptodomex`.

**Comment by CrackerJackCheese** [comment]
Thank you! It worked! Keep up the great work with a-Shell!

**Comment by holzschu** [comment]
That's interesting. The package in `.../<UUID>/a-Shell.app/Library/...` is the one in the app directory, the one you cannot remove (hence the error message). But the package in `.../<UUID>/Library/...` is the one in your home directory, the one that contains the Mac Arm64 libraries. You need to remove the latter (and I thought `pip` would be able to do it without attempting to remove the former).

At this point, I think the only command that works is `rm -rf ~/Library/lib/python3.11/site-packages/Crypto*` , followed by `rm -rf ~/Library/lib/python3.11/site-packages/pycryptodome*`. A bit extreme, but that should work.

**Comment by CrackerJackCheese** [comment]
I figured it out by only removing pycryptodomex.

**Comment by shawnyeager** [comment]
I'm seeing the same problem with the current TestFlight build on iOS 17. Unfortunately, trying to uninstall these packages also fails. 

![image](https://github.com/holzschu/a-shell/assets/980297/f8e2c31b-fe56-400e-b7ca-2bbfc96f71c0)
![image](https://github.com/holzschu/a-shell/assets/980297/78309d72-09aa-41b3-8a47-4cc125a2e576)

**Comment by gaussey** [comment]
> Thanks for the warning. I've run a quick check: the library _is_ included, so the problem must be in `load_pycryptodome_raw_lib`). I'll have a more in-depth look. **Update:** the library is included, but the edited code to load it was not, due to a failed parsing. That's a quick fix, on its way.

Thanks, the update resolved the issue.

**Comment by gaussey** [comment]
Thanks, the update to 1.12.4 resolved the issue.

**Comment by gaussey** [comment]
Thanks the update to 1.12.4 fixed the issue.

**Comment by holzschu** [comment]
@shawnyeager : yes. In a weird Apple logic, the full app can get approved quickly, while the TestFlight for a new version always takes several days. You should get the updated TestFlight version in a few days (and it will be version 1.12.5, since the TestFlight version has to be higher than the AppStore version).

**Comment by shawnyeager** [comment]
Ah, thanks. That’s my fault, then. I had just switched from App Store to TestFlight and mistakenly assumed the latter would be ahead.

**Comment by holzschu** [comment]
Most of the time, you'd be right. But when there's an emergency bug-fix release... I prioritize fast release, so the AppStore goes first.
